signature-specification.md

```diff
@@ -0,0 +1,519 @@
+- Doesn't support multiple signature in one signature envelope.
+
+<details><summary>Example (Click to show details)</summary>
+
+`payload`:
+
+```json
+{ 
+    "subjectManifest": {...},
+    "signedAttrs": {
+        "reserved": {
+            "iat": "2021-06-22T16:02:40.3375379Z",
+            "exp": "2021-09-20T16:02:40.3375379Z",
+            ...
+        },
+        "custom" : {...}
+    }
+}
+```
+
+`footer`:
+
+```json
+{
+    "x5c": [
+        "<Base64(DER(leafCert))>",
+        "<Base64(DER(intermediateCACert))>",
+        "<Base64(DER(rootCert))>"
+    ],
+    "timestamp": "<Base64(TimeStampToken)>"
+}
+```
+
+`Signature Envelope`:
+
+```
+<version>.<purpose>.Base64Url(<payload><Sign(PAE(<version><purpose><payload>))>).Base64Url(<footer>)
+```
+
+- `version`: `v2`, v2 is recommended version.
+- `Purpose`: `public`, public is used for signing.
+- `PAE`: [Pre-Authentication Encoding](https://tools.ietf.org/id/draft-paragon-paseto-rfc-00.html#rfc.section.2.2.1)
+
+</details>
+
+
+### Red Hat's Simple Signing - [Simple Signing](https://www.redhat.com/en/blog/container-image-signing)
+
+It is used by Red Hat for signing container image. Simple signing is *not a signature format* instead defines the payload format which is then signed using PGP to create detached signature. The payload format is adapted by atomic and later by [cosign](https://github.com/sigstore/cosign#payload-formats).
+
+<details><summary>Example (Click to show details)</summary>
+
+`payload`:
+
+```json
+{
+    "critical": {
+        "identity": {
+            "docker-reference": "testing/manifest"
+        },
+        "image": {
+            "Docker-manifest-digest": "sha256:20be...fe55"
+        },
+        "type": "atomic container signature"
+    },
+    "optional": {
+        "creator": "atomic",
+        "timestamp": 1458239713
+    }
+}
+```
+
+</details>
+
+
+## Recommendation
+
+### Signature format
+
+Recommendation is to **use JWS JSON Serialization** because of following reasons:
+
+- Specifications are parts of [RFC](https://datatracker.ietf.org/doc/html/rfc7515#section-3.2) standards.
+- Native support for certificate(s), key id, signed and unsigned attributes.
+- Signing libraries available in Go: [square/go-jose](https://github.com/square/go-jose), [lestrrat-go/jwx](https://github.com/lestrrat-go/jwx), [golang-jwt/jwt](github.com/golang-jwt/jwt) or general JWT libraries to produce / consume a JWS in compact serialization and convert it to / from JSON serialization by custom code.
+- Security related concerns can be addressed with proper implementations.
+
+Concerns with DSSE:
+
+- No published RFC (or draft).
+- New format that is used in few projects.
+- Does not currently support TSA signature and certificate(s) chain as part of the envelope. See [secure-systems-lab/dsse#42](https://github.com/secure-systems-lab/dsse/issues/42),  [secure-systems-lab/dsse#33](https://github.com/secure-systems-lab/dsse/issues/33).

```

Comment:
I'd remove this as a concern since there are various ways to resolve this, including a wrapper around the DSSE envelope, keeping in mind that the OCI artifact layout is already a wrapper around what we're creating here.